bin/calculate_res_gene_depth_qc.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python
+
+import csv
+import argparse
+
+# calculate mean depth coverage for each resistance gene
+def calculate_mean_depth(gene_positions, depth):
+    total_depth = 0
+    total_positions = 0
+
+    for position in range(gene_positions[0], gene_positions[1] + 1):
+        if position in depth:
+            total_depth += depth[position]
+            total_positions += 1
+
+    return total_depth / total_positions
+
+# calculate percent of gene above depth threshold
+def calculate_percent_gene_covered(gene_positions, depth, depth_threshold):
+    covered_positions = 0
+
+    for position in range(gene_positions[0], gene_positions[1] + 1):
+        if position in depth and depth[position] >= depth_threshold:
+            covered_positions += 1
+
+    total_positions = gene_positions[1] - gene_positions[0] + 1
+
+
+    return (covered_positions / total_positions) * 100
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Calculate resistance gene coverage qc metrics.')
+    parser.add_argument('--bed', required=True, help='Path for resistance genes bed file.')
+    parser.add_argument('--depth', required=True, help='Path for tsv file with depth for each position (intermediate mpileup output).')
+    parser.add_argument('--threshold', type=float, default=10, help='min_depth threshold (default: 10).')
+    parser.add_argument('--output',  help='Output CSV file name.')
+    args = parser.parse_args()
```

Comment:
Can we move the argument parsing logic down below the `if __name__ == '__main__':` line, then pass the args into main? ie:

```python
def main(args):
    ...

if __name__ == "__main__":
    parser = argparse.ArgumentParser(description='Calculate resistance gene coverage qc metrics.')
    parser.add_argument('--bed', required=True, help='Path for resistance genes bed file.')
    parser.add_argument('--depth', required=True, help='Path for tsv file with depth for each position (intermediate mpileup output).')
    parser.add_argument('--threshold', type=float, default=10, help='min_depth threshold (default: 10).')
    parser.add_argument('--output',  help='Output CSV file name.')
    args = parser.parse_args()
    main(args)
```

Reply:
[Fixed here](https://github.com/BCCDC-PHL/tbprofiler-nf/pull/14/commits/067b0f6dc533901d7dc196972d189136e8d62de1)